src/caret_analyze/infra/yaml/architecture_reader_yaml.py

```diff
@@ -78,21 +78,49 @@ def _get_value_with_default(
     ):
         return obj.get(key_name, default)
 
-    def _get_publish_topic_names(
+    def _get_publish_topic_infos(
         self,
         node_name: str,
         callback_name: str
-    ) -> list[str]:
+    ) -> list[PublishTopicInfoValue]:
         node_dict = self._get_node_dict(node_name)
         if 'publishes' not in node_dict.keys():
             return []
 
         publishes = self._get_value(node_dict, 'publishes')
-        topic_names = [
-            self._get_value(p, 'topic_name')
-            for p
-            in publishes
-            if callback_name in self._get_value(p, 'callback_names')]
+        topic_names = []
+        for pub in publishes:
+            if callback_name in self._get_value(pub, 'callback_names'):
+                if 'construction_order' in pub.keys():
+                    construction_order = self._get_value(pub, 'construction_order')
+                else:
+                    construction_order = 0
+                pub_info = PublishTopicInfoValue(self._get_value(pub, 'topic_name'),
+                                                 construction_order)
+                topic_names.append(pub_info)
+
+        return topic_names
+
+    def _get_subscribe_topic_infos(
+        self,
+        node_name: str,
+        callback_name: str
+    ) -> list[Any]:
+        node_dict = self._get_node_dict(node_name)
+        if 'subscribes' not in node_dict.keys():
+            return []
+
+        subscribes = self._get_value(node_dict, 'subscribes')
+        topic_names = []
+        for sub in subscribes:
+            if callback_name in self._get_value(sub, 'callback_name'):
+                if 'construction_order' in sub.keys():
+                    construction_order = self._get_value(sub, 'construction_order')
+                else:
+                    construction_order = 0
+                sub_info = ((self._get_value(sub, 'topic_name'), construction_order))
+                topic_names.append(sub_info)
+
```

Comment:
This function appears to be no longer used in the current implementation, so it looks like it should be removed.